examples/amp-first/pages/index.js

```diff
@@ -0,0 +1,253 @@
+                width='64'
+                height='64'
+                alt='demo image'
+              />
+            </template>
+          </amp-list>
+        </section>
+
+        <section>
+          <h3>amp-script</h3>
+          <p>
+            Checkout the{' '}
+            <a href='https://amp.dev/documentation/components/amp-script/'>
+              amp-script
+            </a>{' '}
+            helper here: <code>components/amp/AmpScript.js</code> for embedding
+            custom JavaScript.
+          </p>
+
+          <AmpScript
+            layout='container'
+            src={`${props.host}/static/amp-script/hello.js`}
+          >
+            <button>Hello amp-script!</button>
+          </AmpScript>
+
+          <p>
+            The helper also supports embedding inline scripts. Good to know:
+            Next.js uses{' '}
+            <a href='https://github.com/ampproject/amp-toolbox/tree/master/packages/optimizer'>
+              AMP Optimizer
+            </a>{' '}
+            under the hood, which automatically adds the needed script hashes
+            for{' '}
+            <a href='https://amp.dev/documentation/components/amp-script/#load-javascript-from-a-local-element'>
+              inline amp-scripts
+            </a>
+            .
+          </p>
+          <AmpScript
+            id='hello-world'
+            layout='fixed-height'
+            height='64'
+            script={() => {
+              const btn = document.querySelector('button')
+              btn.addEventListener('click', () => {
+                document.body.textContent = 'Hello World!'
+              })
+            }}
+          >
+            <button>Hello amp-script!</button>
+          </AmpScript>
+        </section>
+      </main>
+    </Layout>
+    <style jsx>{`
+      code,
+      pre {
+        font-family: 'SFMono-Regular', Consolas, 'Liberation Mono', Menlo,
+          Courier, monospace;
+        background: #f2f2f2;
+        padding: 2px 3px;
+        font-size: 13px;
+      }
+      main {
+        margin: 0 auto;
+        max-width: 800px;
+      }
+      main > * + * {
+        margin: 4rem 0.5rem;
+      }
+      .title {
+        text-align: center;
+        padding-top: 4rem;
+      }
+      .hero {
+        display: grid;
+        grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
+        grid-gap: 1rem;
+      }
+      .hero > a {
+        display: block;
+        padding: 1rem;
+        text-align: left;
+        text-decoration: none;
+        background-color: #005af0;
+      }
+      .hero h3 {
+        margin: 0;
+        color: #067df7;
+        color: #fff;
+      }
+      .hero p {
+        margin: 0;
+        color: #fff;
+      }
+    `}</style>
+  </>
+)
+
+// amp-script requires absolute URLs, so we create a property `host` which we can use to calculate the script URL.
+Home.getInitialProps = async ({ req }) => {
+  return { host: `${getProtocol(req)}://${req.headers.host}` }
```

Comment:
`req.headers.host` is not secure in most hosting environments.

Reply:
Good point. Is there a better generic way to get the host or should this be handled via config?

Reply:
Added a comment instead.